cluster/engine.go

```diff
@@ -342,6 +342,13 @@ func (e *Engine) CheckConnectionErr(err error) {
 		return
 	}
 
+	// Docker engine may return 404 when it doesn't recognize a command.
+	// It's not an error from the engine itself. Version validation should be done separately.
+	// This error message is not recorded here to avoid user confusion.
+	if strings.HasPrefix(err.Error(), "404 ") {
```

Comment:
Thanks @nishanttotla for the comment. Actually I want to change it to record only selected few errors, and ignore others because the error itself is not self-contained, i.e., they are not clear without request info. For example, vm7 runs docker engine 1.6.2 and returns error `500 Internal Server Error: image ID '' is invalid` from `docker run hello-world`. It falls in the same category as #1727.  
```
 vm7: xxx.xxx.xxx.xxx:4444
  └ Status: Healthy
  └ Containers: 0
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 4.053 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.19.0-25-generic, operatingsystem=Ubuntu 14.04.3 LTS, storagedriver=aufs
  └ Error: 500 Internal Server Error: image ID '' is invalid
  └ UpdatedAt: 2016-02-02T04:43:19Z
```